src/modules/flow/adafruit/adafruit.c

```diff
@@ -0,0 +1,161 @@
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "adafruit-gen.h"
+#include "sol-flow-internal.h"
+
+#include <sol-spi.h>
+#include <sol-util.h>
+#include <errno.h>
+
+
+struct lcd_strip_data {
+    struct sol_spi *spi;
+    int last_set_pixel;
+    int last_set_color;
+    uint16_t pixel_count;
+    uint16_t pixel_array_length;
+    uint8_t *pixels;
+};
+
+static void
+led_strip_controler_close(struct sol_flow_node *node, void *data)
+{
+    struct lcd_strip_data *mdata = data;
+
+    if (mdata->spi)
+        sol_spi_close(mdata->spi);
+}
+
+static int
+led_strip_controler_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    struct lcd_strip_data *mdata = data;
+    const struct sol_flow_node_type_adafruit_led_strip_options *opts;
+    uint16_t dataBytes;
+    uint8_t latchBytes;
+
+    opts = (const struct sol_flow_node_type_adafruit_led_strip_options *)options;
+
+    SOL_INT_CHECK(opts->pixel_count.val, < 0, -EINVAL);
+    mdata->pixel_count = opts->pixel_count.val;
+
+    dataBytes = mdata->pixel_count * 3;
+    latchBytes = (mdata->pixel_count + 31) / 32;
+    mdata->pixel_array_length = dataBytes + latchBytes;
+
+    mdata->pixels = malloc(mdata->pixel_array_length);
+    SOL_NULL_CHECK(mdata->pixels, -EINVAL);
+    memset(mdata->pixels, 0x80, dataBytes); // Init to RGB 'off' state
+    memset(&mdata->pixels[dataBytes], 0, latchBytes); // Clear latch bytes
+
+    mdata->spi = sol_spi_open(opts->bus.val, opts->chip_select.val);
+    if (mdata->spi) {
+        sol_spi_set_transfer_mode(mdata->spi, 0);
+
+        // Initial reset
+        sol_spi_transfer(mdata->spi, &mdata->pixels[mdata->pixel_count * 3], NULL, latchBytes);
+    }
+
+    return 0;
+}
+
+static void
+_set_pixel_color(struct lcd_strip_data *mdata)
+{
+    uint8_t r, g, b;
+    uint8_t *pixel;
+
+    r = 0xff & (mdata->last_set_color << 16);
+    g = 0xff & (mdata->last_set_color << 8);
+    b = 0xff & mdata->last_set_color;
+
```

Comment:
shouldn't  be:

r = 0xff & (mdata->las_set_color >> 16)
g = 0xff & (mdata->las_set_color >> 8)